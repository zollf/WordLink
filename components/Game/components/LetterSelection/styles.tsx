import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  letterSelection: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
    width: "100%",
    paddingLeft: 20,
    paddingRight: 20,
  },
});

export default styles;