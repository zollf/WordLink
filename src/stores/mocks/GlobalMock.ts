const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
