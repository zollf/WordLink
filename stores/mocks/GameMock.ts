const level: Level = {
  title: 'hello',
  difficulty: 'easy',
  game: {
    start: 'FORTY',
    end: 'FIFTY',
    path: 5,
  },
};

const GameMocks = {
  gameId: 0,
  difficulty: 'EASY',
  inGame: false,
  loaded: false,
  error: false,
  game: level.game,
  gameTitle: 'hello',

  currentWord: level.game.start,
  selectedLetter: '',
  stepNumber: 0,
  path: [level.game.start],
  visited: { FORTY: true },
  completed: false,

  loadGame: jest.fn(),
  startGame: jest.fn(),
  getNewLetters: jest.fn(() => ['A', 'B', 'C']),
  isWord: jest.fn(),
  setSelectedLetter: jest.fn(),
  changeLetter: jest.fn(),
  resetCurrentWord: jest.fn(),
  undo: jest.fn(),
  clearGame: jest.fn(),
};

export default GameMocks;
