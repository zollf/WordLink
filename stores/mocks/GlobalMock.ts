const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
