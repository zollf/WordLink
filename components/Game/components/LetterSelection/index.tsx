import React, { useEffect, useState } from 'react';
import { LayoutAnimation, View } from 'react-native';
import { observer } from 'mobx-react';
import { useStore } from '../../../../hooks';
import Letter from '../Letter';

import styles from './styles';

const LetterSelection = () => {
  const { getNewLetters, setSelectedLetter, selectedLetter, currentWord, completed } = useStore('gameStore');
  const [letters, setLetters] = useState(getNewLetters());

  useEffect(() => { 
    LayoutAnimation.configureNext({ 
      duration: 200, 
      update: { type: 'spring', springDamping: 0.9 }, 
    });
    setLetters(getNewLetters());
  }, [currentWord]);

  return (
    <View style={styles.letterSelection}>
      {!completed && letters.map((letter: string, i: number) => (
        <Letter onPress={() => setSelectedLetter(letter)} color={selectedLetter === letter ? "green" : ""} letter={letter} key={`letterSelection-${letter}-${i}`}/>
      ))}
    </View>
  );
};

export default observer(LetterSelection);