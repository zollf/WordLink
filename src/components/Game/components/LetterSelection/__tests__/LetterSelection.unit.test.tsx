import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import GlobalMock from 'app/src/stores/mocks/GlobalMock';
import GameMock from 'app/src/stores/mocks/GameMock';

import LetterSelection from '..';

const component = (
  <Provider global={GlobalMock} gameStore={GameMock}>
    <LetterSelection />
  </Provider>
);

describe('<LetterSelection />', () => {
  let wrapper: renderer.ReactTestRenderer;
  beforeEach(() => {
    wrapper = renderer.create(component);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click works correctly', () => {
    wrapper.root.findAllByProps({ 'data-test-id': 'selectable-letter' })[0].props.onPress();
    expect(GameMock.setSelectedLetter).toBeCalled();
  });

  it('there is a selected letter', () => {
    GameMock.selectedLetter = 'A';
    wrapper = renderer.create(component);
    expect(wrapper.root.findByProps({ 'data-test-selected': true })).toBeDefined();
  });
});
