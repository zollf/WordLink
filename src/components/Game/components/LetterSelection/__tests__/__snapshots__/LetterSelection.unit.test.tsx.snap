// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LetterSelection /> matches snapshot 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": mockConstructor {},
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.05,
          "shadowRadius": 4,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294177779,
              4293125091,
            ],
            "endPoint": undefined,
            "locations": Array [
              0.1,
              0.8,
            ],
            "startPoint": Array [
              -0.1,
              0.6,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "margin": 3,
            "width": 75,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Raleway_700Bold",
                "fontSize": 40,
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          A
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": mockConstructor {},
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.05,
          "shadowRadius": 4,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294177779,
              4293125091,
            ],
            "endPoint": undefined,
            "locations": Array [
              0.1,
              0.8,
            ],
            "startPoint": Array [
              -0.1,
              0.6,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "margin": 3,
            "width": 75,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Raleway_700Bold",
                "fontSize": 40,
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          B
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": mockConstructor {},
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.05,
          "shadowRadius": 4,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4294177779,
              4293125091,
            ],
            "endPoint": undefined,
            "locations": Array [
              0.1,
              0.8,
            ],
            "startPoint": Array [
              -0.1,
              0.6,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "margin": 3,
            "width": 75,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Raleway_700Bold",
                "fontSize": 40,
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          C
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
